File: app/models/category.py
from sqlalchemy import Column, Integer, String, DateTime
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database import Base

class AssetCategory(Base):
    __tablename__ = "asset_categories"

    category_id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    category_name = Column(String(100), nullable=False)
    description = Column(String(255), nullable=True)
    status = Column(String(20), default="Active", nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    # Relationships
    assets = relationship("Asset", back_populates="category")

File: web-admin/src/types/category.ts
export interface AssetCategory {
  category_id: number;
  category_name: string;
  description?: string;
  status: string;
  created_at?: string;
  updated_at?: string;
}

export interface AssetCategoryCreate {
  category_name: string;
  description?: string;
  status?: string;
}
